pkg/operator/controllers/base/aro_controller.go

```diff
@@ -103,3 +107,31 @@ func (c *AROController) defaultDegraded() *operatorv1.OperatorCondition {
 func (c *AROController) conditionName(conditionType string) string {
 	return c.Name + "Controller" + conditionType
 }
+
+func (c *AROController) IsClusterUpgrading(ctx context.Context) (bool, error) {
+	cv := &configv1.ClusterVersion{}
+	err := c.Client.Get(ctx, types.NamespacedName{Name: "version"}, cv)
+	if err != nil {
+		err = fmt.Errorf("error getting the ClusterVersion: %w", err)
+		c.Log.Error(err)
+		return false, err
+	}
+
+	return version.IsClusterUpgrading(cv), nil
+}
```

Comment:
ok - just a thought I had - is this singular case an appropriate one for us to set the ARO Operator into Progressing until all MCs are successfully applied and validated? If I understand correctly, this would put the upgrade on "pause" until we are 100% reconciled.

Reply:
It's this case and also for the full deprecation of DNSMasq. What is the concern about the operator status remaining in progressing for a prolonged period of time? Is it that it may obfuscate other issues?